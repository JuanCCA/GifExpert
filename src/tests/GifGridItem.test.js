import { shallow } from "enzyme";
import {GifGridItem} from "../components/GifGridItem"
describe('Testing GifGridItems', () => {

    const title = 'Titulo de prueba'
    const url = 'https://api.bimi.com/aklsfk'
    const wrapper = shallow( <GifGridItem title={title} url={url} />);
    
    test('Should create component correctly', () => {
        
        expect(wrapper).toMatchSnapshot()

    })
    

    test('Should have a p with title', () => {
        
        const p = wrapper.find(`p`).text();
        expect(p).toBe(title)

    })

    test('should have image and alt same', () => {
        

        const img = wrapper.find('img');
        const src = img.prop('src');
        const alt = img.prop('alt');
        expect(src).toBe(url)
        expect(alt).toBe(title);

    })

    test('should have fade class', () => {
        
        const fade = "animate__fadeInLeftBig"

        const div = wrapper.find('div')

        const classes = div.prop('className')

        const existsClass =  classes.includes(fade)

        expect(existsClass).toBe(true)


    })
    
    
    

})
