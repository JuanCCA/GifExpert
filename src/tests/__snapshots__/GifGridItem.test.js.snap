// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItems Should create component correctly 1`] = `
<div
  className="card animate__animated animate__fadeInLeftBig"
>
  <img
    alt="Titulo de prueba"
    src="https://api.bimi.com/aklsfk"
  />
  <p>
    Titulo de prueba
  </p>
</div>
`;
